src/plugins/ui_actions/public/types.ts

```diff
@@ -20,12 +20,21 @@
 import { Action } from './actions/action';
 import { TriggerInternal } from './triggers/trigger_internal';
 
-export type TriggerRegistry = Map<string, TriggerInternal<any>>;
-export type ActionRegistry = Map<string, Action>;
-export type TriggerToActionsRegistry = Map<string, string[]>;
+export type TriggerRegistry = Map<TriggerId, TriggerInternal<any>>;
+export type ActionRegistry = Map<ActionId, Action<any>>;
+export type TriggerToActionsRegistry = Map<TriggerId, ActionId[]>;
 
-export type TriggerId = string;
+const DEFAULT_TRIGGER = '';
+
+export type TriggerId = typeof DEFAULT_TRIGGER | keyof TriggerContextMapping;
```

Comment:
This is the magic that made this happen!  Now TriggerId type picks up all the added keys in TriggerContextMapping!

Reply:
Is `typeof DEFAULT_TRIGGER` part required? 
Seems like because `DEFAULT_TRIGGER` is keyof `TriggerContextMapping` - should work without it.

Reply:
Good call, removed!